src/Uno.UWP/Storage/ApplicationData/ApplicationDataCompositeValue.cs

```diff
@@ -0,0 +1,210 @@
+	/// Creates and initializes a new, initially empty, instance of the object.
+	/// </summary>
+	public ApplicationDataCompositeValue()
+	{
+	}
+
+	/// <summary>
+	/// Creates and initializes a new instance of the with the specified dictionary.
+	/// </summary>
+	/// <param name="dictionary">Dictionary.</param>
+	/// <exception cref="ArgumentNullException">Thrown if parameter is null.</exception>
+	internal ApplicationDataCompositeValue(Dictionary<string, object?> dictionary)
+	{
+		_dictionary = dictionary ?? throw new ArgumentNullException(nameof(dictionary));
+	}
+
+	/// <summary>
+	/// Occurs when the observable map has changed.
+	/// </summary>
+	public event MapChangedEventHandler<string, object?>? MapChanged;
+
+	/// <summary>
+	/// Gets the number of items contained in the property set.
+	/// </summary>
+	public uint Size => (uint)_dictionary.Count;
+
+	/// <summary>
+	/// Gets teh number of items contained in the property set.
+	/// </summary>
+	public int Count => _dictionary.Count;
+
+	/// <summary>
+	/// Gets a value indicating whether the collection is read-only.
+	/// </summary>
+	public bool IsReadOnly => false;
+
+	/// <summary>
+	/// Adds an item to the property set.
+	/// </summary>
+	/// <param name="key">The key to insert.</param>
+	/// <param name="value">The value to insert.</param>
+	public void Add(string key, object? value)
+	{
+		_dictionary.Add(key, value);
+		MapChanged?.Invoke(this, new MapChangedEventArgs(CollectionChange.ItemInserted, key));
+	}
+
+	/// <summary>
+	/// Indicates whether the property set has an item with the specified key.
+	/// </summary>
+	/// <param name="key">Key.</param>
+	/// <returns>True if found.</returns>
+	public bool ContainsKey(string key) => _dictionary.ContainsKey(key);
+
+	/// <summary>
+	/// Removes a key from the property set.
+	/// </summary>
+	/// <param name="key">The key to remove.</param>
+	/// <returns>True if the key was found.</returns>
+	public bool Remove(string key)
+	{
+		var result = _dictionary.Remove(key);
+		if (result)
+		{
+			MapChanged?.Invoke(this, new MapChangedEventArgs(CollectionChange.ItemRemoved, key));
+		}
+
+		return result;
+	}
+
+	/// <summary>
+	/// Tries to get the value for the specified key.
+	/// </summary>
+	/// <param name="key">The key to retrieve.</param>
+	/// <param name="value">The value correspodning with the key.</param>
+	/// <returns>True if found.</returns>
+	public bool TryGetValue(string key, out object? value)
+	{
+		if (key is null)
+		{
+			throw new ArgumentNullException(nameof(key));
+		}
+		return _dictionary.TryGetValue(key, out value);
+	}
+
+	/// <summary>
+	/// Gets or sets a value for the specified key.
+	/// </summary>
+	/// <param name="key">The key.</param>
+	/// <returns>Value.</returns>
+	public object? this[string key]
+	{
+		get => _dictionary[key];
+		set
+		{
+			// Add or update and raise map changed accrodingly			
+			if (_dictionary.TryGetValue(key, out var existingValue))
+			{
+				if (value != existingValue)
```

Comment:
It looks like this is a reference equality, which may be incorrect in certain scenarios? e.g, boxed integers?